### ISSUE TITLE
Allow elastic search _bulk operations to set the index name in the URL 

### ISSUE THREAD

**Issue by Tommmster** [akka/alpakka#1843]
### Short description

The elasticsearch Sink will fail sending messages when `rest.action.multi.allow_explicit_index` is set to `true` in the `elasticsearch.yml`. Moreover, this sink provides no functionality to set the index in the URL.

### Details
`_bulk` operations do not include the index name in the URL, but add the `_index` field to [each document](https://github.com/akka/alpakka/blob/master/elasticsearch/src/main/scala/akka/stream/alpakka/elasticsearch/impl/ElasticsearchFlowStage.scala#L143). However, the elasticsearch cluster may not allow this. (see https://www.elastic.co/guide/en/elasticsearch/reference/current/url-access-control.html). If this is the case, the cluster will reject the bulk operation.

The [`ElasticsearchFlowStage` constructor](https://github.com/akka/alpakka/blob/master/elasticsearch/src/main/scala/akka/stream/alpakka/elasticsearch/impl/ElasticsearchFlowStage.scala#L27) has all the information needed to handle this situation. By adding a field to the `ElasticsearchWriteSettings` to indicate whether explicit index names are allowed, the `sendBulkUpdateRequest` method can change it's behaviour accordingly (that is, using the /indexname/_bulk endpoint and *not* setting the `_index` field in the payload)

#### Proposal
* Add a `multiAllowExplicitIndex` field to the `ElasticsearchWriteSettings` case class, with a default value of `true` (keeps current behaviour)

* Create two `WriteMessage[T,C] => Seq[(String, JsString)]` functions. One will provide the `("_index" -> JsString(indexname))` tuple, the other one will not. Keep the appropiate one in a field.

* Modify this signature
```
private def sendBulkUpdateRequest(messages: immutable.Seq[WriteMessage[T,C]]): Unit
```
to this one
```
private def sendBulkUpdateRequest(messages: immutable.Seq[WriteMessage[T,C]], 
                                                            bulkSharedFields: WriteMessage[T,C] => Seq[(String, JsString)]): Unit
```

and provide the appropiate function 

* Add a field indicating the endpoint to pass to the `RestClient`. Possible values are `/_bulk` (current) and `/$indexName/_bulk` 

* In the `ElasticsearchFlowStage[T,C]` constructor, read this property and determine

**Comment by 2m** [comment]
Thanks for the detailed analysis. Approach sounds good. Reviewing the PR.

**Comment by ennru** [comment]
Implemented with #2326
